### ISSUE TITLE
electron-rebuild failed for ibm_db

### ISSUE THREAD

**Issue by Uzair-Fasih** [electron/electron-rebuild#324]
Hello,

I am trying to connect to a DB2 database in my electron app using **ibm_db** package.
I get the following error when running electron-rebuild

**OS:** Windows 10
**Electron:** 2.0.4

```
× Rebuild Failed

An unhandled error occurred inside electron-rebuild
gyp info it worked if it ends with ok
gyp info using node-gyp@5.0.5
gyp info using node@10.15.0 | win32 | x64
gyp info find Python using Python version 3.7.2 found at "C:\Users\Dell\AppData\Local\Programs\Python\Python37-32\python.exe"
gyp http GET https://electronjs.org/headers/v2.0.18/iojs-v2.0.18.tar.gz
gyp http 200 https://electronjs.org/headers/v2.0.18/iojs-v2.0.18.tar.gz
gyp http GET https://electronjs.org/headers/v2.0.18/SHASUMS256.txt
gyp http GET https://electronjs.org/headers/v2.0.18/win-x86/iojs.lib
gyp http GET https://electronjs.org/headers/v2.0.18/win-arm64/iojs.lib
gyp http GET https://electronjs.org/headers/v2.0.18/win-x64/iojs.lib
gyp http 200 https://electronjs.org/headers/v2.0.18/SHASUMS256.txt
gyp http 200 https://electronjs.org/headers/v2.0.18/win-x64/iojs.lib
gyp http 403 https://electronjs.org/headers/v2.0.18/win-arm64/iojs.lib
gyp WARN install got an error, rolling back install
gyp http 200 https://electronjs.org/headers/v2.0.18/win-x86/iojs.lib
gyp ERR! configure error
gyp ERR! stack Error: 403 status code downloading arm64 iojs.lib
gyp ERR! stack     at Request.<anonymous> (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\electron-rebuild\node_modules\node-gyp\lib\install.js:335:22)
gyp ERR! stack     at Request.emit (events.js:187:15)
gyp ERR! stack     at Request.onRequestResponse (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\request\request.js:1066:10)
gyp ERR! stack     at ClientRequest.emit (events.js:182:13)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:556:21)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:442:20)
gyp ERR! stack     at TLSSocket.emit (events.js:182:13)
gyp ERR! stack     at addChunk (_stream_readable.js:283:12)
gyp ERR! stack     at readableAddChunk (_stream_readable.js:264:11)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Dell\\Desktop\\Workspace\\Atlas\\node_modules\\electron-rebuild\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=2.0.18" "--arch=x64" "--dist-url=https://electronjs.org/headers" "--build-from-source"
gyp ERR! cwd C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\ibm_db
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok

Failed with exit code: 1

Error: gyp info it worked if it ends with ok
gyp info using node-gyp@5.0.5
gyp info using node@10.15.0 | win32 | x64
gyp info find Python using Python version 3.7.2 found at "C:\Users\Dell\AppData\Local\Programs\Python\Python37-32\python.exe"
gyp http GET https://electronjs.org/headers/v2.0.18/iojs-v2.0.18.tar.gz
gyp http 200 https://electronjs.org/headers/v2.0.18/iojs-v2.0.18.tar.gz
gyp http GET https://electronjs.org/headers/v2.0.18/SHASUMS256.txt
gyp http GET https://electronjs.org/headers/v2.0.18/win-x86/iojs.lib
gyp http GET https://electronjs.org/headers/v2.0.18/win-arm64/iojs.lib
gyp http GET https://electronjs.org/headers/v2.0.18/win-x64/iojs.lib
gyp http 200 https://electronjs.org/headers/v2.0.18/SHASUMS256.txt
gyp http 200 https://electronjs.org/headers/v2.0.18/win-x64/iojs.lib
gyp http 403 https://electronjs.org/headers/v2.0.18/win-arm64/iojs.lib
gyp WARN install got an error, rolling back install
gyp http 200 https://electronjs.org/headers/v2.0.18/win-x86/iojs.lib
gyp ERR! configure error
gyp ERR! stack Error: 403 status code downloading arm64 iojs.lib
gyp ERR! stack     at Request.<anonymous> (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\electron-rebuild\node_modules\node-gyp\lib\install.js:335:22)
gyp ERR! stack     at Request.emit (events.js:187:15)
gyp ERR! stack     at Request.onRequestResponse (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\request\request.js:1066:10)
gyp ERR! stack     at ClientRequest.emit (events.js:182:13)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:556:21)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:442:20)
gyp ERR! stack     at TLSSocket.emit (events.js:182:13)
gyp ERR! stack     at addChunk (_stream_readable.js:283:12)
gyp ERR! stack     at readableAddChunk (_stream_readable.js:264:11)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Dell\\Desktop\\Workspace\\Atlas\\node_modules\\electron-rebuild\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=2.0.18" "--arch=x64" "--dist-url=https://electronjs.org/headers" "--build-from-source"
gyp ERR! cwd C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\ibm_db
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok

Failed with exit code: 1
    at SafeSubscriber._error (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\spawn-rx\lib\src\index.js:267:84)
    at SafeSubscriber.__tryOrUnsub (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.error (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:156:26)
    at Subscriber._error (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:92:26)
    at Subscriber.error (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:72:18)
    at MapSubscriber.Subscriber._error (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:92:26)
    at MapSubscriber.Subscriber.error (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:72:18)
    at SafeSubscriber._next (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\spawn-rx\lib\src\index.js:242:65)
    at SafeSubscriber.__tryOrUnsub (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.next (C:\Users\Dell\Desktop\Workspace\Atlas\node_modules\rxjs\internal\Subscriber.js:143:22)
```

Any help is much appreciated.
😃

**Comment by wavifu** [comment]
You can manually update node-gyp to version 6.0.1 to fix this. 

To do this, go to your **package-lock.json**, find **electron-rebuild** block. In the **requires** block , remove *node-gyp* entry there, and in the **dependencies** block, add 

`"node-gyp": {"version": "6.0.1"}`

Then go to terminal, remove **node_modules** directory, and re-run 

`npm install`

It will install node-gyp v6.0.1 instead of v5.0.5 for electron-rebuild. You then can use it to rebuild your code.

**Comment by malept** [comment]
This is fixed in 1.8.7.
